### PR TITLE
Add yoast promo in seo settings banner

### PR DESCRIPTION
**Pull request by robertsreberski**
## Proposed changes:

* Modify the Banner component to support icons without circle design
* Add Yoast promo banner to SEO settings

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Checkout this branch `add/yoast-promo-traffic-block`
* Run your docker env
* Go to `/wp-admin/admin.php?page=jetpack#/traffic`
* See that "Search engine optimization" block promotes Yoast Free:
CleanShot 2023-03-22 at 18 57 16@2x

* Ensure CTA link is `https://yoa.st/yoast-jetpack-boost?domain=<your_site>`
* Go to "/wp-admin/plugins.php" and install (and activate) Yoast plugin
* Go to `/wp-admin/admin.php?page=jetpack#/traffic`
* See that "Search engine optimization" block promotes Yoast Premium:
CleanShot 2023-03-22 at 18 58 21@2x

* Ensure CTA link is `https://yoa.st/yoast-upgrade-jetpack-boost?domain=<your_site>`
* A12s - reach out to me to install Yoast Premium plugin
* Go to `/wp-admin/admin.php?page=jetpack#/traffic`
* The ad should not be displayed anymore

CleanShot 2023-03-22 at 18 59 54@2x

### CODE REVIEW

**Review by jeherve — COMMENTED** [review]
This tests well, but I spotted a few issues:

1. When you use Yoast SEO on your site, we already have a notice to let you know our feature was automatically disabled and your SEO is now handled by the Yoast SEO plugin:
Screenshot 2023-03-24 at 12 25 17
Adding the new promotion banner on top of that feels odd for some reason. It's a lot of information at once, I think that may be confusing for some. I wonder if we could do something better, it may be worth asking someone on Jetpack Design to take a look  maybe?

2. When the SEO tools are already active, the banner doesn't fit exactly under the card:

Screenshot 2023-03-24 at 12 28 58

**Review by IanRamosC — DISMISSED** [review]
The code itself looks good and the functionality tests well. ✅ 
I don't have any comments other than what @jeherve mentioned above.


> Ensure CTA link is https://yoa.st/yoast-jetpack-boost?domain=<your_site>

The final URL is https://yoast.com/wordpress/plugins/seo/#shortlink=yoast-jetpack-boost

>Ensure CTA link is https://yoa.st/yoast-upgrade-jetpack-boost?domain=<your_site>

The final URL is https://yoast.com/reasons-to-upgrade/#shortlink=yoast-upgrade-jetpack-boost

**Comment by robertsreberski** [comment]
I fixed the weird spacing when SEO options are enabled, thanks @jeherve, for spotting it!

With issue (1) - we've decided to change the text a bit, and we'll wait for confirmation from Yoast regarding it. We'll handle the copy changes in the separate PR.

**Comment by robertsreberski** [comment]
@IanRamosC regarding urls - that's true, testing plan is outdated as initially we've made direct links to yoast - but later we decided to use jetpack redirects. Links should be as following:

- https://jetpack.com/redirect/?source=jetpack-boost-yoast-free&query=domain%3D<your_site>&site=<your_site>
- https://jetpack.com/redirect/?source=jetpack-boost-yoast-upgrade&query=domain%3D<your_site>&site=<your_site>

File: projects/plugins/jetpack/_inc/client/traffic/style.scss
Comment (IanRamosC) [inline projects/plugins/jetpack/_inc/client/traffic/style.scss]: It seems like `.jp-seo-social-previews` already has enough specificity to not need `!important`.

Comment (robertsreberski) [reply projects/plugins/jetpack/_inc/client/traffic/style.scss]: Oh, I was following the way it's done in other blocks, I'll remove redundant `!important` now, thanks!

**Review by IanRamosC — APPROVED** [review]
LGTM!

**Comment by sybrew** [comment]
You may want to add The SEO Framework to that reduced list of conflicting plugins before merging this feature. 

I'm not sure what happened to this list https://github.com/Automattic/jetpack/blob/d37390724e5513f4068c1ee54b109dd4cfc0168a/projects/plugins/jetpack/class.jetpack.php#L240-L252.

**Comment by sybrew** [comment]
You may want to add The SEO Framework to that [reduced list of conflicting plugins](https://github.com/Automattic/jetpack/blob/5a264de4e131a358dbcdfe82d95bcf569adce8e1/projects/plugins/jetpack/_inc/client/traffic/seo.jsx#L21-L54) before merging this feature. 

I'm not sure what happened to this list https://github.com/Automattic/jetpack/blob/d37390724e5513f4068c1ee54b109dd4cfc0168a/projects/plugins/jetpack/class.jetpack.php#L240-L252.

Regardless, this is a bad move -- it's mimicking Awesome Motive, and they're the largest thundercunts on the internet. I'm not happy with the current state of WordPress for which you're responsible. You're reaping the benefits of being the biggest, but it will crash if you continue. Sr. Growth Engineer...? More like Sr. Devourer. 

Expect hell from the community after this merge.

**Comment by jeherve** [comment]
@sybrew Thanks for the feedback. We're still iterating on this, so will definitely take your feedback into account.

To address your other question:

> You may want to add The SEO Framework to that [reduced list of conflicting plugins](https://github.com/Automattic/jetpack/blob/5a264de4e131a358dbcdfe82d95bcf569adce8e1/projects/plugins/jetpack/_inc/client/traffic/seo.jsx#L21-L54) before merging this feature.
> I'm not sure what happened to this list

That list is still active. They control different parts of our feature set and UI. I'm happy to add SEO Framework to the list, for sure. I just created #29728 to take care of that. 👍

**Comment by sybrew** [comment]
Thank you, but that PR does not requite the unindemnifiable action of the antitrust movement the merge of this PR created.

Jetpack is a "[Featured](https://wordpress.org/plugins/browse/featured/)" piece of crippling serviceware riddled with dark patterns, only created to shovel money toward Automattic, ultimately operating under a 501(c)3 Foundation that should benefit nobody but the people that use the software. Promoting trialware (and its premium upgrade) makes this entire operation fall under the legislated market; you cannot hide behind "FOSS."

I advise you to undo the results of the agreement you have with Yoast B.V. You may also want to look into my arguments here: https://twitter.com/SybreWaaijer/status/1639609372488736768.

**Pull request merged**
